### PR TITLE
fix cpu/gpu oed's eig&ewma tests

### PR DESCRIPTION
**Pull request by fehiepsi**
This PR addresses bugs announced in #1373 and #1418.

We get those bugs because:
+ EmwaLog uses old style of `torch.autograd.Function`, which might not be supported anymore (it still works in some cases but there will be many edge cases around the old style).
+ EmwaLog is initialized with `torch.tensor`, so GPU tests will fail.

### CODE REVIEW

File: pyro/contrib/oed/eig.py [hunk 1]
```diff
@@ -334,7 +335,22 @@ def loss_fn(design, num_particles):
     return loss_fn
 
 
-class EwmaLog(torch.autograd.Function):
+class _EwmaLogFn(torch.autograd.Function):
```
Comment (fritzo) [inline pyro/contrib/oed/eig.py]: Can you add a dosctring explaining how this class differs from the usual `torch.log()`?

Comment (fehiepsi) [reply pyro/contrib/oed/eig.py]: We have doc string at EwmaLog though. I didn't change it so it might not show up in git diff.
